### DISCUSSION TITLE
How to Create a Type-Safe, Flexible useCookie Hook in TypeScript | Seraj Vahdati

### DISCUSSION THREAD

**Discussion by giscus** [seraj/blog#25] [General]
# How to Create a Type-Safe, Flexible useCookie Hook in TypeScript | Seraj Vahdati

a hook for adding, removing, and modifying cookies

https://seraj.me/blog/useCookie-hook

**Comment by dwjohnston** [comment]
It should be noted that this solution will only respond to cookie value changes if those changes are made via the hook. If a change is made directly, or as a result of API responses, the hook will not be aware of the change.

**Reply by dwjohnston** [reply]
Possibly adding something like [CookieChangeEvent](https://developer.mozilla.org/en-US/docs/Web/API/CookieChangeEvent) - could help here.

**Reply by dwjohnston** [reply]
Is not supported by Firefox though! Boo! :(

**Reply by dwjohnston** [reply]
I have an updated guide if you want your hook to responsive to all cookies here, including a polyfill for Firefox and Safari. https://blacksheepcode.com/posts/responsive_cookies_for_react
